src/vos/vos_aggregate.c

```diff
@@ -389,7 +389,34 @@ vos_agg_sv(daos_handle_t ih, vos_iter_entry_t *entry,
 	 */
 	if (agg_param->ap_max_epoch == 0 ||
 	    agg_param->ap_max_epoch == entry->ie_epoch) {
-		agg_param->ap_max_epoch = entry->ie_epoch;
+
+		switch (entry->ie_dtx_state) {
+		case DTX_ENT_COMMITTED:
+			/* Highest epoch is committed, keep it. */
+			agg_param->ap_max_epoch = entry->ie_epoch;
+			break;
+		case DTX_ENT_UNCOMMITTED:
+			/*
+			 * Highest epoch is uncommitted, skip it and continue
+			 * checking on next lower epoch.
+			 */
+			D_DEBUG(DB_EPC, "Skip uncommitted at epoch:"DF_X64"\n",
+				entry->ie_epoch);
+			break;
+		case DTX_ENT_ABORTED:
```

Comment:
So before this patch being landed, current master may merge an abort entry into some valid visible entry? If that is true, why we never found related data corruption in former CI tests or just because we never verify the data after VOS aggregation with aborted entries?

Reply:
There is no merge for SV, but before the patch, aggregation treats all entries as 'committed'.